### ISSUE TITLE
Feature request: 1D CNN, i.e. keras.layer.Conv1d

### ISSUE THREAD

**Issue by liuyxpp** [LuxDL/Lux.jl#709]
Conv1d is very useful for time series, can we have it in Lux?

**Comment by avik-pal** [comment]
It is just `Conv((kernel_size,))` where kernel_size is an integer

**Comment by liuyxpp** [comment]
Would it be possible to add this into the built-in layers? It is very helpful/friendly for someone like me (novice in deep learning but want to reproduce paper results mostly written in PyTorch, JAX or Keras).

**Comment by avik-pal** [comment]
It is already built-in, right? Instead of Conv1d, simply write Conv. That is the same reason we don't have Conv2d/3d/..., all of them simply differ by how kernels are defined, i.e. a 1-Tuple is Conv1d, 2-tuple is Conv2d, and so on.

**Comment by liuyxpp** [comment]
Yes, I get your concerns but I still think it is no harm to make the layer more explicit. Anyway, I have one more question: if my data is (nfeatures, ntimesteps, nbatch), i.e. multivariate time series. I shall first reshape it to (nfeatures, ntimesteps, 1, nbatch), and then define the layer and chain them together as follows
```julia
rs = ReshapeLayer((nfeatures, ntimesteps, 1))
conv1 = Conv((1, kernel_size), 1=>1)
model = Chain(rs, conv1)
```
to perform 1D CNN for each feature?

**Comment by avik-pal** [comment]
That will perform convolution over the timesteps rather than features, since you are defining a 2D operation with 1 channel.

I think you might want to do `Conv((kernel_size,), ntimesteps => ntimesteps)` if you need to convolve over the features.

**Comment by liuyxpp** [comment]
As I understand the literature, 1D CNN for time series is indeed do convolution over timesteps. So my wording "perform 1D CNN for each feature" means "perform 1D CNN for each time series (one for each feature)."

In your solution, it seems not quite follow the doc which says

> Image data should be stored in WHCN order (width, height, channels, batch).

It states that 4D data is mandatory? and

> this layer expects as input an array with `ndims(x) == N + 2`

It states that the channel dim and thebatch dim are mandatory no matter the feature dims?
